src/log/Entry.h

```diff
@@ -90,6 +90,7 @@ class ConcreteEntry : public Entry {
     str.assign(strv.begin(), strv.end());
     return *this;
   }
+  ConcreteEntry(ConcreteEntry& e) = default;
```

Comment:
>  In order to call capacity(), ConcreteEntry needed its copy constructor back.

Did you mean `set_capacity()`?  It would be really weird if `m_recent.capacity()` call required a copy constructor.

Reply:
hmm, makes sense. it's as the `circular_buffer` might need to reallocate a new buffer for resizing itself. in that case, `circular_buffer` could use copy constructor or move-copy constructor for moving the elements from the old place to the new place using `move_if_noexcept`. also, per https://www.boost.org/doc/libs/1_79_0/doc/html/circular_buffer/implementation.html#circular_buffer.implementation.move_emulation_and_rvalues , because `ConcreteEntry` contains a moveable member variable `str`, we can potentially benefit from its move constructor, probably we can mark the move constructor noexcept so `circular_buffer` can use it instead of the more expensive copy ctor? like:
```c++
oncreteEntry(ConcreteEntry&& e) noexcept : Entry(e), str(std::move(e.str)) {}
```